lib/msf/core/post/common.rb

```diff
@@ -27,18 +27,18 @@ def rhost
     return nil unless session
 
     case session.type
-    when 'meterpreter'
+    when /meterpreter/
       session.sock.peerhost
-    when 'shell'
+    when /(shell|powershell)/
       session.session_host
     end
```

Comment:
Let's break this out into explicit cases uses strings instead of regexs.

```suggestion
    case session.type
    when 'meterpreter'
      session.sock.peerhost
    when 'powershell', 'shell'
      session.session_host
    end

Reply:
okay I'll update this, thanks :+1: